### PR TITLE
API: make cache flushing a public, documented operation

### LINKED ISSUE #3003: API: make flush cache work as documented or update docs
Whatever is the better (not easier!) option to do.

### PR DESCRIPTION
**Pull request by zeha**
Note that this changes the URL, and only allows for exact flushing now.

Fixes #3003.

### CODE REVIEW

**Comment by zeha** [comment]
Rebased, fixed. Recursor tests fail until #2997 is merged.

File: pdns/ws-auth.cc [hunk 1]
```diff
@@ -1160,16 +1160,23 @@ static void apiServerSearchData(HttpRequest* req, HttpResponse* resp) {
   resp->setBody(doc);
 }
 
-void apiServerFlushCache(HttpRequest* req, HttpResponse* resp) {
+void apiServerCacheFlush(HttpRequest* req, HttpResponse* resp) {
   if(req->method != "PUT")
     throw HttpMethodNotAllowedException();
 
+  string name = req->getvars["domain"];
+  if (name.empty() || name[name.size()-1] != '.') {
```
Comment (pieterlexis) [inline pdns/ws-auth.cc]: maybe make this 2 checks so you can give a more specific error message?

File: regression-tests.api/test_Cache.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from test_helper import ApiTestCase, is_auth, is_recursor
+
+
+class Servers(ApiTestCase):
```
Comment (pieterlexis) [inline regression-tests.api/test_Cache.py]: I don't see an actual test to see if the cache indeed has been flushed? Will this be done when cache-snooping is implemented in the API?

Comment (zeha) [inline regression-tests.api/test_Cache.py]: Right now, during tests, the cache is always empty, and I don't want to add cache-prefilling now. When cache-snooping is in, we can add proper tests for that; for now I'll go with believing the server it did whatever it thinks it needs doing.

File: pdns/ws-auth.cc (hunk 1 above)
Comment (zeha) [inline pdns/ws-auth.cc]: I think I'll go with making this the same generic 'unable to parse' error actually.

**Comment by zeha** [comment]
I've refactored the DNSName construction on all api sites so it behaves the same everywhere, and that got rid of the stupid empty/dot check in the flush functions.

**Pull request merged**